### DISCUSSION TITLE
Is there a pagination with numbered results?

### DISCUSSION THREAD

**Discussion by sri1985** [datenstrom/community#325] [Ask a question] [answered]
Is there any plugin that enables Numbered Results. Currently the results are given as "Next", Previous. If there is a numbered result like: 1 2 3 4 5 ...10 etc,, it would be really great! This is really needed in the Blog page.

Thanks

**Comment by markseuffert** [comment]
I am sorry, not that I know. How about making your own pagination snippet? Have a look at file `system/themes/snippets/pagination.php` You can use the functions `getPaginationNumber()` and `getPaginationCount()` to create a numbered result. See [API](https://datenstrom.se/yellow/help/api-for-developers) for more information.

**Comment by StephanWeinhold** [comment, marked as answer]
@sri1985 I needed this too, so there you go: https://gist.github.com/StephanWeinhold/8c97efe6862abb7a246ad94dbf4a09fb

**Comment by markseuffert** [comment]
Thank you. Can you elaborate a little bit on the template coding style and what could be improved?

**Comment by StephanWeinhold** [comment]
I prefer a more readable style (line breaks after PHP-tags; curly brackets instead of colons; spaces between statement and parentheses; line break before an `else`; semicolons to mark the end of a line; etc.). **But** I know that maintaining templates is way more complicated then. So I totally understand why you chose this style - pardon my hint! And thanks for asking!

**Comment by markseuffert** [comment]
Makes a lot of sense. You are welcome to point out problems and alternatives. I think it's helpful to hear what people think and give us feedback. No matter how may work you put into software design, you seldom get it perfect the first iterations. Let's make it work, then make it better. 😄
